From 94c10ca95536d614d994e9abe9ba139b5ecfc552 Mon Sep 17 00:00:00 2001
From: Shawn Landden <shawnlandden@gmail.com>
Date: Fri, 8 Jun 2012 04:31:20 -0700
Subject: [PATCH] fix test.rb to work with gem2deb

 test.rb |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/test.rb b/test.rb
index 17ea4ce..d879f17 100755
--- a/test.rb
+++ b/test.rb
@@ -29,6 +29,15 @@ formats = [
             "kctest.rb misc '%s'",
            ]
 
+# For Debian: gem2deb builds both 1.8 and 1.9.1 in succession, leading this file to be the wrong version
+# remove this so that installed versions get included
+# feel free to remove this block so that you build according to the docs
+begin
+  File.unlink("kyotocabinet.so")
+rescue Errno::ENOENT
+ # second time file wont be there
+end
+
 system("rm -rf casket*")
 rubycmd = Config::CONFIG["bindir"] + "/" + RbConfig::CONFIG['ruby_install_name']
 all = confs.size * formats.size
-- 
1.7.9.5

